### PR TITLE
Add YAML docs for Task

### LINKED ISSUE #396: Docs: Add/Update schema docs for yaml + templates
Add/Update schema docs for yaml + templates

### PR DESCRIPTION
**Pull request by ehotinger**
**Purpose of the PR:**

- Add YAML docs for Task

**Fixes #396**

### CODE REVIEW

**Comment by ehotinger** [comment]
Open to feedback on this, expect it to be relatively iterative. /cc @sajayantony 

@mnltejaswini  + @shahzzzam  can you also suggest improvements to docs for secrets. I know that area is mostly still a WIP so it can be a rough skeleton for now.

**Comment by shahzzzam** [comment]
finally :D

File: docs/task.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+#### startDelay
+
+Defines the number of seconds to wait before executing the container.
+
+* Optional
+* Type: `int`
+
+#### retries
+
+The number of retries to attempt if a container fails its execution.
```
Comment (northtyphoon) [inline docs/task.md]: Can you elaborate how we determine containers fails? eg, return code is not 0

File: docs/task.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+* Optional
+* Type: `string`
+
+### step
+
+An object with the following properties:
+
+| Property | Type | Required | Nullable |
+|----------|------|----------|----------|
+| [id](#id) | `string` | Optional | Yes |
```
Comment (northtyphoon) [inline docs/task.md]: What does Nullable mean? Especially for int or bool type. I think we probably need a default value column.

File: docs/task.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
+# Task Schema
```
Comment (northtyphoon) [inline docs/task.md]: add a version in the header

File: docs/task.md (hunk 2 above)
Comment (ehotinger) [reply docs/task.md]: Removed, added default value

File: docs/task.md (hunk 3 above)
Comment (ehotinger) [reply docs/task.md]: Done

File: docs/task.md (hunk 1 above)
Comment (ehotinger) [reply docs/task.md]: Done

**Pull request merged**